docs/reference/core/runtime.md

```diff
@@ -82,20 +83,243 @@ We can easily access the default `Runtime` to run an effect:
+
+```scala
+timestamp=2022-09-01T08:07:47.870219Z level=INFO thread=#zio-fiber-6 message="Started effectful workflow to customize runtime configuration" location=<empty>.MainApp.effectfulConfiguration file=ZIOApp.scala line=8
+Application started!
+Application is about to exit!
+```
+
+## Top-level Runtime Configuration
+
+When we write a ZIO application using the `ZIOAppDefault` trait, a default top-level runtime is created and used to run the application automatically under the hood. Further, we can customize the rest of the ZIO application by providing locally scoped configuration layers using [`provideXYZ` operations](#configuring-runtime-by-providing-configuration-layers) or [`bootstrap` layer](#configuring-runtime-using-bootstrap-layer).
+
+This is usually sufficient for lots of ZIO applications, but it is not always the case. There are cases where we want to customize the runtime of the entire ZIO application from the top level.
+
+In such cases, we need to create a top-level runtime by unsafely running the configuration layer to convert that configuration to the `Runtime` by using the `Runtime.unsafe.fromLayer` operator:
+
+```scala mdoc:invisible
+import zio._
+val layer = ZLayer.empty
+```
+
+```scala mdoc:compile-only
+val runtime: Runtime[Any] =
+  Unsafe.unsafe { implicit unsafe =>
+    Runtime.unsafe.fromLayer(layer)
+  }
+```
+
+Let's try a fully working example:
+
+```scala mdoc:compile-only
+import zio._
+
+object MainApp extends ZIOAppDefault {
+
+  // In a real-world application we might need to implement a `sl4jlogger` layer
+  val addSimpleLogger: ZLayer[Any, Nothing, Unit] =
+    Runtime.addLogger((_, _, _, message: () => Any, _, _, _, _) => println(message()))
+
+  val layer: ZLayer[Any, Nothing, Unit] =
+    Runtime.removeDefaultLoggers ++ addSimpleLogger
+
+  override val runtime: Runtime[Any] =
+    Unsafe.unsafe { implicit unsafe =>
+      Runtime.unsafe.fromLayer(layer)
+    }
+
+  def run = ZIO.log("Application started!")
+}
+```
+
+This is useful when we want to install a custom monitoring or supervisor from the very beginning of the application.
+
+:::caution
+Keep in mind that only the "bootstrap" layer of applications will be combined when we compose two ZIO applications. Therefore, when we compose two ZIO programs, top-level runtime configurations won't be integrated.
+:::
+
+Another use-case of top-level runtimes is when we want to integrate our ZIO application inside a legacy application:
+
+```scala mdoc:compile-only
+import zio._
+
+object MainApp {
+  val sl4jlogger: ZLogger[String, Any] = ???
+
+  def legacyApplication(input: Int): Unit = ???
+
+  val zioWorkflow: ZIO[Any, Nothing, Int] = ???
 
-Sometimes we need to create a custom `Runtime` with a user-defined environment.
+  def zioApplication(): Int =
+      Unsafe.unsafe { implicit unsafe =>
```

Comment:
It could be useful to separate the creation of the `Runtime` from the use of it to execute a workflow here since often the reason to create a top level runtime is to be able to execute multiple different workflows from the top level.